cibersort/cibersort.xml

```diff
@@ -19,120 +19,130 @@
-
-  # Determine colors and graph
-  ##summedData = summedData[summedData[,"Sample"]=="$sample_id"]
-  colors = getColors(cellTypes, map=groups, level=level)
-  graph(summedData, level, colors)
+    # Sort the cell types for consistent visualization
+    #colnames(summedData) = c("CellType", "Sample", "EstimatedPercent")
+    summedData[,level] = as.character(summedData[,level])
+    summedData[,level] = factor(summedData[,level], levels=cellTypes)
+    summedData[order(summedData[,level]),] 
+    data = summedData
 }
-#end if
+
+# Determine colors and then graph
+colors = getColors(cellTypes, map=groups, level=level)
+graph(data, level, writeName, colors=colors)
         </configfile> 
     </configfiles>
 
     <inputs>
         <param name="signature_file" type="data" format="tsv" label="Gene Signature File" />
         <param name="expression_file" type="data" format="tsv" label="Cohort Gene Expression"
                help="Hugo-named gene expression file" />
-        <param name="sample_ids" type="text" optional="false" label="Sample ID" />
+
+        <section name="Plots">
+            <param name="sample_ids" type="text" optional="true" label="Sample ID(s)"
+                   help="Comma separated list, or all samples if not provided" />
+            <param name="sequential_axis" type="select" optional="false" label="Select x-axis for sequential plots">
+                <option value="Sample">Group By Sample</option>
+                <option value="CellType">Gourp By Cell Type</option>
+            </param>
+        </section>
+
         <conditional name="mapping">
-            <param name="provided" type="select" label="Provide Mapping File">
-                <option value="TRUE">Provide File</option>
-                <option value="FALSE">Omit File</option>
+            <param name="cell_types" type="select" label="Cell Type Definition">
+                <option value="custom">Use Custom Cell Types</option>
+                <option value="default">Use Default CIBERSORT Cell Types</option>
             </param>
-            <when value="TRUE">
-                <param name="map_file" type="data" format="tsv" label="Mapping File (TSV)"/>
-                <param name="map_colors" type="boolean" default="true" label="Use Colors From Map File"/>
-                <param name="resolutions" type="select" multiple="true" display="checkboxes" optional="false"
-                       label="Cell Type Resolution" help="Select one or both of the cell type definition levels">
+            <when value="custom">
+                <param name="map_file" type="data" format="tsv" label="TSV containing cell type map and optional color specification"/>
+                <param name="map_colors" type="boolean" checked="true" label="Use Colors From Map File"/>
+                <param name="resolution" type="select" multiple="false" optional="false" 
+                       label="Cell Type Resolution" help="Select one of the custom cell mappings">
```

Comment:
Can options be extracted from column names of map file? Ask Qiang for help

Reply:
Example:
```
<param name="colnames"  type="data_column" use_header_names="true" data_ref="inputfile" label="Select target column(s):"/>
```
And docs: https://docs.galaxyproject.org/en/latest/dev/schema.html#data-column